### ISSUE TITLE
My ember build is very slow

### ISSUE THREAD

**Issue by dmtask** [ember-cli/ember-cli#8783]
Hi,

my ember build is after the ember-cli update from 2.18.0 to 3.11.0 very slow. What can i do for fix it? Is my node version to old?

Build Times:
```
Build successful (24232ms)

Slowest Nodes (totalTime => 5% )              | Total (avg)         
----------------------------------------------+---------------------
Bundler (1)                                   | 5151ms              
Babel: frontend (2)                           | 3411ms (1705 ms)    
Package /assets/vendor.js (1)                 | 1450ms              
broccoli-persistent-filter:TemplateCom... (7) | 1449ms (207 ms)     
Babel: @ember-data/store (2)                  | 1404ms (702 ms)     
SassCompiler (1)                              | 1308ms              
Babel: ember-test-helpers (2)                 | 1266ms (633 ms)
```

Output from `ember version --verbose && npm --version && yarn --version`:
```
ember-cli: 3.11.0
http_parser: 2.8.0
node: 8.16.0
v8: 6.2.414.77
uv: 1.23.2
zlib: 1.2.11
ares: 1.10.1-DEV
modules: 57
nghttp2: 1.33.0
napi: 4
openssl: 1.0.2r
icu: 64.2
unicode: 12.1
cldr: 35.1
tz: 2019a
os: darwin x64
6.4.1
1.1.0
```

My package.json:
```
{
    "name": "frontend",
    "version": "0.0.0",
    "description": "SD core Frontend",
    "license": "MIT",
    "author": "eoceo",
    "directories": {
        "doc": "doc",
        "test": "tests"
    },
    "repository": "",
    "scripts": {
        "build": "ember build",
        "start": "ember server",
        "test": "ember test"
    },
    "dependencies": {
        "ember-cli-babel": "^7.8.0",
        "ember-cli-htmlbars": "^3.1.0",
        "ember-cli-shims": "^1.2.0",
        "ember-concurrency": "^1.0.0",
        "ember-data": "3.12.0",
        "ember-file-upload": "^2.4.4",
        "ember-js-pdf": "^0.4.2",
        "ember-responsive": "^3.0.5",
        "ember-source": "3.12.0",
        "ember-wormhole": "^0.5.2",
        "fabulous-table": "^1.0.5",
        "jquery-confirm": "^3.3.2",
        "node-sass": "^4.7.2",
        "print-js": "^1.0.61"
    },
    "devDependencies": {
        "broccoli-asset-rev": "^3.0.0",
        "ember-ajax": "^5.0.0",
        "ember-array-helper": "^5.0.0",
        "ember-cable": "0.8.1",
        "ember-cli": "^3.11.0",
        "ember-cli-app-version": "^3.1.3",
        "ember-cli-dependency-checker": "^3.2.0",
        "ember-cli-head": "0.4.1",
        "ember-cli-htmlbars-inline-precompile": "^2.1.0",
        "ember-cli-ifa": "^0.7.0",
        "ember-cli-mirage": "^1.1.0",
        "ember-cli-moment-shim": "^3.5.0",
        "ember-cli-qunit": "^4.3.0",
        "ember-cli-rails-addon": "0.10.0",
        "ember-cli-sass": "^10.0.1",
        "ember-cli-spinner": "0.0.9",
        "ember-cli-sri": "^2.1.0",
        "ember-cli-test-loader": "^2.2.0",
        "ember-cli-uglify": "^3.0.0",
        "ember-cookies": "^0.4.0",
        "ember-custom-actions": "3.2.0",
        "ember-export-application-global": "^2.0.0",
        "ember-highcharts": "0.5.4",
        "ember-i18n": "^5.1.0",
        "ember-ionicons": "^0.1.1",
        "ember-load-initializers": "^2.0.0",
        "ember-moment": "^7.5.0",
        "ember-resolver": "^5.2.1",
        "highcharts": "^6.2.0",
        "loader.js": "^4.6.0"
    },
    "engines": {
        "node": ">= 8.9.1 < 9"
    },
    "private": true
}
```

**Comment by jlami** [comment]
I'm having similar problems. Though I might have to add that I am running on windows.

During some performance analysis on this system I saw that **ember-cli-babel** was my main culprit. If you want to do some of your own traces you can look [here](https://github.com/ember-cli/ember-cli/blob/master/docs/perf-guide.md).

Maybe you can run `npm ls ember-cli-babel` to see if your setup is comparable?
When I run this command I see that a lot of my addons use different versions of ember-cli-babel. And only one version is 'deduped'. All the addons that have a different version will add extra overhead to load there own version.

I believe this could be fixed by a feature that somehow caches the addon by version in ember-cli itself. Or we can (ab)use yarn [resolutions](https://yarnpkg.com/lang/en/docs/selective-version-resolutions/) to overrule the versions to the highest one. But I'm not sure if there are any downsides to this.
This relates to https://github.com/ember-cli/rfcs/pull/112 but I can't find a new related rfc in https://github.com/emberjs/rfcs. So maybe @rwjblue can weigh in here.

**Comment by dmtask** [comment]
Ok, thanks for your answer. I will check this.

**Comment by jlami** [comment]
What I found out after testing with yarn resolutions and going back and forth between a single ember-cli-babel was that yarn was always significantly faster than npm which I used before.

It turns out that yarn would install all the latest versions because there was no _yarn.lock_ file to start from. While npm would keep some packages back because the version in _package-lock.json_ would still match. So removing my _package-lock.json_ and doing a clean `npm install` gave me the same speedup that `yarn install` would give me.

I'm not sure what the yarn behavior is with yarn.lock, but I would guess that it would be the same as npm in holding back packages that still match. So you might have the same luck as me that removing yarn.lock will result in a better installation.

Lesson to learn: package-lock.json/yarn.lock is great for stable installations, but not so good for major updates.
